components/director/internal/domain/formationconstraint/service.go

```diff
@@ -150,6 +151,20 @@ func (s *service) ListByFormationTemplateIDs(ctx context.Context, formationTempl
 
 // Delete deletes formation constraint by id
 func (s *service) Delete(ctx context.Context, id string) error {
+	formationTemplateConstraintReferences, err := s.formationTemplateConstraintReferenceRepo.ListByConstraintID(ctx, id)
```

Comment:
Can we just use exists?

Reply:
We could use Exist - but then we wouldnt know which are the Formation Templates from which the constraint has to be potentialy detached